distributed/client.py

```diff
@@ -3753,6 +3753,14 @@ def upload_file(self, filename, load: bool = True):
         >>> client.upload_file('mylibrary.egg')  # doctest: +SKIP
         >>> from mylibrary import myfunc  # doctest: +SKIP
         >>> L = client.map(myfunc, seq)  # doctest: +SKIP
+        >>>
+        >>> # Where did that file go? Use `dask_worker.local_directory`.
+        >>> def where_is_mylibrary(dask_worker):
+        >>>     path = pathlib.Path(dask_worker.local_directory) / 'mylibrary.egg'
+        >>>     assert path.exists()
+        >>>     return path
```

Comment:
nit: You may want to return the string representation of the path here to avoid issues with `WindowsPath` not being implemented on UNIX systems (the same probably goes for `PosixPath` on Windows). I haven't tested this though.

Reply:
oh ya, that'd be better. https://github.com/dask/distributed/pull/8313/commits/25110d762af66922d732e6764df0b30e6815cacc Thanks!